src/t8_schemes/t8_scheme_common_cxx.hxx

```diff
@@ -0,0 +1,631 @@
+    T8_ASSERT (t8_element_is_valid (elem2));
+    int level1 = t8_element_level (elem1);
+    int level2 = t8_element_level (elem2);
+    int maxlevel = SC_MAX (level1, level2);
+    t8_linearidx_t id1 = t8_element_get_linear_id (elem1, maxlevel);
+    t8_linearidx_t id2 = t8_element_get_linear_id (elem2, maxlevel);
+    return id1 < id2 ? -1 : (id1 > id2 ? 1 : (level1 < level2 ? -1 : (level1 > level2 ? 1 : 0)));
+  }
+
+  virtual t8_gloidx_t
+  t8_element_count_leaves_from_root (int level) const override
+  {
+    t8_element_t *root;
+    t8_element_new (1, &root);
+    t8_element_root (root);
+    t8_gloidx_t num_leaves = t8_element_count_leaves (root, level);
+    t8_element_destroy (1, &root);
+    return num_leaves;
+  }
+  /** Return SC_ABORT for all face functionality */
+
+  /** Compute the number of faces of a given element.
+   * \param [in] elem The element.
+   * \return          The number of faces of \a elem.
+   */
+  virtual int
+  t8_element_num_faces (const t8_element_t *elem) const
+  {
+    SC_ABORT ("not yet implemented");
+  }
+
+  /** Compute the maximum number of faces of a given element and all of its
+   *  descendants.
+   * \param [in] elem The element.
+   * \return          The maximum number of faces of \a elem and its descendants.
+   */
+  virtual int
+  t8_element_max_num_faces (const t8_element_t *elem) const
+  {
+    SC_ABORT ("not yet implemented");
+  }
+
+  /** Return the number of children of an element's face when the element is refined.
+   * \param [in] elem   The element whose face is considered.
+   * \param [in] face   A face of \a elem.
+   * \return            The number of children of \a face if \a elem is to be refined.
+   */
+  virtual int
+  t8_element_num_face_children (const t8_element_t *elem, int face) const
+  {
+    SC_ABORT ("not yet implemented");
+  }
+
+  /** Return the corner number of an element's face corner.
+   * Example quad: 2 x --- x 3
+   *                 |     |
+   *                 |     |   face 1
+   *               0 x --- x 1
+   *      Thus for face = 1 the output is: corner=0 : 1, corner=1: 3
+   *
+   * \param [in] element  The element.
+   * \param [in] face     A face index for \a element.
+   * \param [in] corner   A corner index for the face 0 <= \a corner < num_face_corners.
+   * \return              The corner number of the \a corner-th vertex of \a face.
+   *
+   * The order in which the corners must be given is determined by the eclass of \a element:
+   * LINE/QUAD/TRIANGLE:  No specific order.
+   * HEX               :  In Z-order of the face starting with the lowest corner number.
+   * TET               :  Starting with the lowest corner number counterclockwise as seen from
+   *                      'outside' of the element.
+   */
+  /* TODO: Prism order, Pyramid order. */
+  virtual int
+  t8_element_get_face_corner (const t8_element_t *element, int face, int corner) const
+  {
+    SC_ABORT ("not yet implemented");
```

Comment:
This is just a single comment from glancing at this.
Not a full review.

I would suggest changing the abort message.
"not yet implemented" implied that this function (namely t8_scheme_common_c::t8_element_get_face_corner) will get implemented in the future.
But your concept is that it never will be implemented.

Maybe simply "not implemented" is better. Or even a
"Needs to be implemented in derived class" or smth. to deliver the message clearly.